sin_test.go

```diff
@@ -0,0 +1,60 @@
+package govec
+
+import (
+	"math"
+	"testing"
+)
+
+func TestV2F_Sin(t *testing.T) {
+	v1 := V2F[float64]{X: 0, Y: math.Pi / 2}
+	v2 := v1.Sin()
+
+	if v2.X != 0 || v2.Y != 1 {
+		t.Errorf("V2F Sin failed")
+	}
+}
+
+func TestV2F_SinInPlace(t *testing.T) {
+	v1 := V2F[float64]{X: 0, Y: math.Pi / 2}
+	v1.SinInPlace()
+
+	if v1.X != 0 || v1.Y != 1 {
+		t.Errorf("V2F SinInPlace failed")
+	}
+}
+
+func TestV3F_Sin(t *testing.T) {
+	v1 := V3F[float64]{X: 0, Y: math.Pi / 2, Z: 3 * math.Pi / 2}
+	v2 := v1.Sin()
+
+	if v2.X != 0 || v2.Y != 1 || v2.Z != -1 {
+		t.Errorf("V3F Sin failed")
+	}
+}
+
+func TestV3F_SinInPlace(t *testing.T) {
+	v1 := V3F[float64]{X: 0, Y: math.Pi / 2, Z: 3 * math.Pi / 2}
+	v1.SinInPlace()
+
+	if v1.X != 0 || v1.Y != 1 || v1.Z != -1 {
+		t.Errorf("V3F SinInPlace failed")
+	}
+}
+
+func TestV2I_Sin(t *testing.T) {
+	v1 := V2I[int16]{X: 0, Y: 1}
+	v2 := v1.Sin()
+
+	if v2.X != 0 || v2.Y != 0.8414709848078965 {
```

Comment:
Let's fix 2 things here:
1. Let's not hard-code those crazy sin/con/tan numbers. Instead we can calculate them with `math.Sin`. I know it seems redundant but this way we can make sure it's always ok even if the implementation of the vector operator suddenly would use some other library.
2. Let's not compare float values using `==` and `!=`. You can use the `almostEqual` helper method to compare values with precision of ~0.000001 https://github.com/Xkonti/govec/blob/e0bae7a671ca43dd45a15dd8994e2537b4b92e7e/testUtils.go#L8-L11

### Example
```go
if !almostEqual(v2.X,  math.Sin(0), 0.000001) || !almostEqual(v2.Y, math.Sin(1), 0.000001) { ... }
```

Let's apply that to all tests here.